pkg/helm/handlers/handler_test.go

```diff
@@ -290,7 +330,217 @@ func TestHelmHandlers_HandleGetChart(t *testing.T) {
 			if response.Body.String() != tt.expectedResponse {
 				t.Errorf("response body not matching expected is %s and received is %s", tt.expectedResponse, response.Body.String())
 			}
+		})
+	}
+}
 
+func TestHelmHandlers_HandleGetReleaseHistory(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedResponse    string
+		history             []*release.Release
+		expectedContentType string
+		error
+		httpStatusCode int
+	}{
+		{
+			name:                "chart release history should error out when there is error from helm",
+			error:               errors.New("Chart path is invalid"),
+			expectedResponse:    `{"error":"Failed to list helm release history: Chart path is invalid"}`,
+			httpStatusCode:      http.StatusBadGateway,
+			expectedContentType: "application/json",
+		},
+		{
+			name:                "Get successful release history",
+			expectedResponse:    `[{"name":"test","version":1},{"name":"test","version":2}]`,
+			history:             fakeReleaseHistory,
+			httpStatusCode:      http.StatusOK,
+			expectedContentType: "application/json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handlers := fakeHelmHandler()
+			handlers.getReleaseHistory = fakeGetReleaseHistory(tt.history, tt.error)
+
+			request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
```

Comment:
this test does check if we decode request params properly.

Reply:
this test does not require request body since it's a GET request to get release history

Reply:
Done